src/core/zowe/core_for_zowe_sdk/config_file.py

```diff
@@ -340,3 +339,135 @@ def load_secure_props(self) -> None:
                 f" with error '{error_msg}'",
                 SecurePropsNotFoundWarning,
             )
+
+    def _retrieve_credential(self, service_name: str) -> Optional[str]:
+        """
+        Retrieve the credential from the keyring or storage.
+        If the credential exceeds the maximum length, retrieve it in parts.
+        Parameters
+        ----------
+        service_name: str
+            The service name for the credential retrieval
+        Returns
+        -------
+        str
+            The retrieved  encoded credential
+        """
+        encoded_credential = keyring.get_password(service_name, constants["ZoweAccountName"])
+
+        if encoded_credential is None and sys.platform == "win32":
+            # Filter or suppress specific warning messages
+            warnings.filterwarnings("ignore", message="^Retrieved an UTF-8 encoded credential")
+            # Retrieve the secure value with an index
+            index = 1
+            temp_value = keyring.get_password(f"{service_name}-{index}", f"{constants['ZoweAccountName']}-{index}")
+            while temp_value is not None:
+                if encoded_credential is None:
+                    encoded_credential = temp_value
+                else:
+                    encoded_credential += temp_value
+                index += 1
+                temp_value = keyring.get_password(f"{service_name}-{index}", f"{constants['ZoweAccountName']}-{index}")
+                
+        if encoded_credential is not None and encoded_credential.endswith("\0"):
+            encoded_credential = encoded_credential[:-1]
+
+        try:
+            return encoded_credential.encode('utf-16le').decode()
+        except (UnicodeDecodeError, AttributeError):
+            # The credential is not encoded in UTF-16
+            return encoded_credential
```

Comment:
I believe we only need this logic on Windows, as other platforms are less weird and don't encode the credentials as UTF-16.